src/main/java/com/facebook/openwifirrm/ucentral/informationelement/Country.java

```diff
@@ -0,0 +1,171 @@
+ */
+public class Country {
+	private static final Logger logger = LoggerFactory.getLogger(Country.class);
+
+	/** Defined in 802.11 */
+	public static final int TYPE = 7;
+
+	public static class CountryInfo {
+		/**
+		 * The lowest channel number in the CountryInfo.
+		 */
+		public final int firstChannelNumber;
+		/**
+		 * The maximum power, in dBm, allowed to be transmitted.
+		 */
+		public final int maximumTransmitPowerLevel;
+		/**
+		 * Number of channels this CountryInfo applies to. E.g., if First
+		 * Channel Number is 2 and Number of Channels is 4, this CountryInfo
+		 * describes channels 2, 3, 4, and 5.
+		 */
+		public final int numberOfChannels;
+
+		/**
+		 * Constructor. This constructor is currently never used, but it is
+		 * required by Java since at least one member variable is marked final.
+		 */
+		public CountryInfo(
+			int firstChannelNumber,
+			int maximumTransmitPowerLevel,
+			int numberOfChannels
+		) {
+			this.firstChannelNumber = firstChannelNumber;
+			this.maximumTransmitPowerLevel = maximumTransmitPowerLevel;
+			this.numberOfChannels = numberOfChannels;
+		}
+
+		/** Parse CountryInfo from the appropriate Json object. */
+		public static CountryInfo parse(JsonObject contents) {
+			final int firstChannelNumber =
+				contents.get("First Channel Number").getAsInt();
+			final int maximumTransmitPowerLevel = contents
+				.get("Maximum Transmit Power Level (in dBm)")
+				.getAsInt();
+			final int numberOfChannels =
+				contents.get("Number of Channels").getAsInt();
+			return new CountryInfo(
+				firstChannelNumber,
+				maximumTransmitPowerLevel,
+				numberOfChannels
+			);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(
+				firstChannelNumber,
+				maximumTransmitPowerLevel,
+				numberOfChannels
+			);
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			}
+			if (obj == null) {
+				return false;
+			}
+			if (getClass() != obj.getClass()) {
+				return false;
+			}
+			CountryInfo other = (CountryInfo) obj;
+			return firstChannelNumber == other.firstChannelNumber &&
+				maximumTransmitPowerLevel == other.maximumTransmitPowerLevel &&
+				numberOfChannels == other.numberOfChannels;
+		}
+
+		@Override
+		public String toString() {
+			return "CountryInfo [firstChannelNumber=" + firstChannelNumber +
+				", maximumTransmitPowerLevel=" + maximumTransmitPowerLevel +
+				", numberOfChannels=" + numberOfChannels + "]";
+		}
+	}
+
+	/**
+	 * Each constraint is a CountryInfo describing tx power constraints on
+	 * one or more channels, for the current country.
+	 */
+	public final List<CountryInfo> constraints;
+
+	/**
+	 * Constructor. This constructor is currently never used, but it is
+	 * required by Java since at least one member variable is marked final.
+	 */
+	public Country(List<CountryInfo> countryInfos) {
+		this.constraints = Collections.unmodifiableList(countryInfos);
+	}
+
```

Comment:
Maybe I'm wrong, but it looks weird to me that an object can only be constructed by a list of objects of its internal class. Feel free to push me back

Reply:
Hm, good point... I don't know of a better way. At least, since the nested class is static and public, I don't think it's a big deal. Anyone else have thoughts?

Reply:
I'm fine with the current way, but just wondering if there's a better way. Open to any idea here.

Reply:
Zhiqian also mentioned it may be fine since the inner class is static